3_javascript/javascript2/week3/script.js

```diff
@@ -0,0 +1,217 @@
+})
+
+//9.
+
+function jokeCreator (shouldTellFunnyJoke, logFunnyJoke, logBadJoke) {
+     if (shouldTellFunnyJoke === true) {
+        logFunnyJoke()
+     } else { 
+        logBadJoke()
+     }
+}
+
+function logFunnyJoke () {
+    console.log("What is the area at the Danish/German border called? The DaneGer zone!")
+}
+
+function logBadJoke () {
+    console.log("If you go to a Scandinavian bakery you could Finnish a Swedish Danish. Norway I'd make this up!")
+}
+
+jokeCreator(true, logBadJoke, logFunnyJoke);
+jokeCreator(false, logBadJoke, logFunnyJoke);
+
+
+//2. Function as a variable
+//1. (Create an array with 3 items. All items should be functions. Iterate through the array and call all the functions.)
+const functionArray = [
+    function item1 () {
+        console.log("Hello function array 1")
+    },
+    function item2 () {
+        console.log("Hello function array 2")
+    },
+    function item3 () {
+        console.log("Hello function array 3")
+    }
+]
+
+for (let index = 0; index < functionArray.length; index++) {
+    const element = functionArray[index];
+    element()
+}
+
+//2. (Create a function as a const and try creating a function normally. Call both functions.)
+const firstFunction = function () {
+    return 5  ; 
+}
+
+function simpleFunction() {
+return 10 ;
+}
+
+const first = firstFunction();
+const second = simpleFunction();
+
+console.log(first);
+console.log(second);
+
+//3. (Create an object that has a key whose value is a function. Try calling this function.)
+const valueFunction = function ( message) {
+    console.log(message)
+}
+
+const obj = {valueFunction : "I love coding"}
+
+valueFunction(obj.valueFunction)
+
+//3. The fastest presser in this realm (stuck in this)
+
+ domLoader()
+
+let sCounter = 0;
+let lCounter = 0;
+let gameDuration = 0;
+
+const durationInput = document.getElementById("durationInput");
+const startGameButton = document.getElementById("btn-3");
+const resultDiv = document.getElementById("result");
+
+        function startGame() {
+            const userInput = durationInput.value;
+            if (userInput) {
+                gameDuration = parseInt(userInput, 10);
+                resultDiv.innerHTML = `Game started! Type 'S' or 'L'.`;
+
+                setTimeout(() => {
+                    resultDiv.innerHTML = `Game over! You typed 'S' ${sCounter} times and 'L' ${lCounter} times.`; 
+                    determineWinner();
+                }, gameDuration * 1000);
+            }
+        }
+        startGameButton.addEventListener("click", startGame);
+
+        document.addEventListener("keydown", function (event) {
+            const key = event.key;
+            if (key === "S" || key === "s") {
+                sCounter++;
+            } else if (key === "L" || key === "l") {
+                lCounter++;
+            }
+        });
+
+        function determineWinner() {
+            if (sCounter > lCounter) {
+                resultDiv.innerHTML += "<br>S wins!";
```

Comment:
it would be clinner to use a paragraph <p> here, as well as for the game started and game over text.

Reply:
I am not able to understand how should I use a paragraph in it, I think I am confuse a bit.Can you help me .